packages/services/src/infrastructure/api/hiro/hiro-stacks-api.client.ts

```diff
@@ -1,38 +1,145 @@
+} from '@stacks/stacks-blockchain-api-types';
 import axios from 'axios';
 
+import { DEFAULT_LIST_LIMIT } from '@leather.io/constants';
+
 import { HttpCacheService } from '../../cache/http-cache.service';
 import { HttpCacheTimeMs } from '../../cache/http-cache.utils';
+import { RateLimiterService, RateLimiterType } from '../../rate-limiter/rate-limiter.service';
+import { NetworkSettingsService } from '../../settings/network-settings.service';
+import { hiroApiRequestsPriorityLevels } from './hiro-request-priorities';
 
-interface HiroPaginatedResponse<T> {
-  limit: number;
-  offset: number;
-  total: number;
-  results: T;
-}
-
-export type HiroAddressTransactionsResponse = HiroPaginatedResponse<Transaction>;
+export type HiroAddressTransactionsListResponse = AddressTransactionsListResponse;
+export type HiroAddressBalanceResponse = AddressBalanceResponse;
+export type HiroMempoolTransactionListResponse = MempoolTransactionListResponse;
+export type HiroFtMetadataResponse = FtMetadataResponse;
 
 export interface HiroStacksApiClient {
-  getAddressTransactions(
+  getAddressBalances(address: string, signal?: AbortSignal): Promise<HiroAddressBalanceResponse>;
+  getAddressConfirmedTransactions(
     address: string,
     signal?: AbortSignal
-  ): Promise<HiroAddressTransactionsResponse>;
+  ): Promise<HiroAddressTransactionsListResponse>;
+  getAddressMempoolTransactions(
+    address: string,
+    signal?: AbortSignal
+  ): Promise<HiroMempoolTransactionListResponse>;
+  getFungibleTokenMetadata(
+    principal: string,
+    signal?: AbortSignal
+  ): Promise<HiroFtMetadataResponse>;
 }
 
-export function createHiroStacksApiClient(cache: HttpCacheService) {
-  async function getAddressTransactions(address: string, signal?: AbortSignal) {
-    return await cache.fetchWithCache<HiroAddressTransactionsResponse>(
-      ['hiro-stacks-address-transactions', address],
+export function createHiroStacksApiClient(
+  cache: HttpCacheService,
+  network: NetworkSettingsService,
+  limiter: RateLimiterService
+): HiroStacksApiClient {
+  async function getAddressBalances(address: string, signal?: AbortSignal) {
+    return await cache.fetchWithCache(
+      ['hiro-stacks-get-address-balances', address, network.getConfig().chain.stacks.chainId],
       async () => {
-        const res = await axios.get(``, { signal });
+        const res = await limiter.add(
+          RateLimiterType.HiroStacks,
+          () =>
+            axios.get<HiroAddressBalanceResponse>(
+              `${network.getConfig().chain.stacks.url}/extended/v1/address/${address}/balances`,
+              { signal }
+            ),
+          {
+            priority: hiroApiRequestsPriorityLevels.getAccountBalance,
+            signal,
+            throwOnTimeout: true,
+          }
+        );
+        return res.data;
+      },
+      { ttl: HttpCacheTimeMs.twoMinutes }
+    );
+  }
+
+  async function getAddressConfirmedTransactions(address: string, signal?: AbortSignal) {
```

Comment:
Great work on this! This gets a lot of the data we will need for `Activity`. I wonder could we also add BTC transaction data to`infrastructure`?